File: app/src/main/java/org/bspb/smartbirds/pro/ui/map/BspbMapFragment.java
package org.bspb.smartbirds.pro.ui.map;

import com.google.android.gms.maps.SupportMapFragment;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.Bean;
import org.androidannotations.annotations.EFragment;
import org.bspb.smartbirds.pro.events.EEventBus;
import org.bspb.smartbirds.pro.events.MapAttachedEvent;
import org.bspb.smartbirds.pro.events.MapDetachedEvent;

/**
 * Created by dani on 14-11-17.
 */
@EFragment
public class BspbMapFragment extends SupportMapFragment {

    @Bean
    EEventBus eventBus;

    @AfterViews
    public void init() {
        eventBus.post(new MapAttachedEvent());
    }

    @Override
    public void onDetach() {
        super.onDetach();
        eventBus.post(new MapDetachedEvent());
    }
}

File: app/src/main/kotlin/org/bspb/smartbirds/pro/ui/map/BspbMapFragment.kt
package org.bspb.smartbirds.pro.ui.map

import android.os.Bundle
import android.view.View
import com.google.android.gms.maps.SupportMapFragment
import org.bspb.smartbirds.pro.events.EEventBus
import org.bspb.smartbirds.pro.events.MapAttachedEvent
import org.bspb.smartbirds.pro.events.MapDetachedEvent

class BspbMapFragment : SupportMapFragment() {

    private val eventBus: EEventBus = EEventBus.getInstance()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        eventBus.post(MapAttachedEvent())
    }

    override fun onDetach() {
        super.onDetach()
        eventBus.post(MapDetachedEvent())
    }
}